.gitignore

```diff
@@ -42,3 +42,10 @@ build/Railroad.jar
 
 # Local disk buffers for "roachprod logs" command
 /*.logs
+
+# Generated code.
+*.pb.go
+*.pb.gw.go
+pkg/kv/kvclient/rangefeed/mocks_generated.go
```

Comment:
We have .gitignores within specific packages, perhaps these could go there?
```
$  find . -type f | grep gitignore
[...]
./sql/parser/.gitignore
./sql/opt/optgen/lang/support/vscode/cockroachdb.optgen-1.0.0/.gitignore
./sql/opt/xform/.gitignore
./sql/opt/norm/.gitignore
./sql/opt/.gitignore
./sql/opt/memo/.gitignore
```

Reply:
Sorry, what's the request here? That I add new `.gitignore`s for the packages referenced here and add these files to those `.gitignore`s? If it's up to me, I really prefer having this stuff in one place, but I'll make that change if we're adamant about it.

Reply:
Not at all adamant -- excited to see this go in!